### DISCUSSION TITLE
Measuring distances on a model?

### DISCUSSION THREAD

**Discussion by watson-e-and** [google/model-viewer#4270] [Q&A]
I'd like users of my model to be able to measure distances on the model, kind of like on Google Earth where you can drop pins to measure the distance between points. Is this something that model-viewer can do? Is there a sample anywhere? 

I used the sample at modelviewer.dev to add dimension lines to my model (from the Annotation section), which is very nice for giving overall context to the model, but adding something that's more flexible would be wonderful.

**Comment by elalish** [comment]
Yes, you can use [positionAndNormalFromPoint](https://modelviewer.dev/docs/index.html#entrydocs-annotations-methods-positionAndNormalFromPoint) to find the model-space coordinates of a user-clicked point - with two of those you can calculate the distance and show it in a DOM element. You can even make that DOM element a hotspot so that it follows the model like a dimension line if you want.

**Reply by watson-e-and** [reply]
This seems great! Is there any sample code for positionAndNormalFromPoint? I wasn't able to find any from following the links. This is one of my first projects in Javascript and HTML so I'm still learning my way around how the pieces fit together.

**Reply by elalish** [reply]
No, the closest we have is `materialFromPoint` [here](https://modelviewer.dev/examples/scenegraph/#pickMaterialExample), which is broadly similar. You can also see how we use it in our editor (`packages/space-opera`), but that's probably not a great place to start if you're new to JS.
